unlock-app/src/__tests__/services/walletService.test.js

```diff
@@ -734,6 +734,67 @@ describe('WalletService', () => {
       })
     })
 
+    describe('partialWithdrawFromLock', () => {
+      let lock
+      let account
+
+      beforeEach(() => {
+        lock = '0xd8c88be5e8eb88e38e6ff5ce186d764676012b0b'
+        account = '0xdeadbeef'
+      })
+
+      it('should invoke sendTransaction with the right params', () => {
+        expect.assertions(3)
+        const data = '' // mock abi data for partialWithdraw
+
+        walletService._sendTransaction = jest.fn()
+
+        const ContractClass = class {
+          constructor(abi, address) {
+            expect(abi).toBe(LockContract.abi)
+            expect(address).toBe(lock)
+            this.methods = {
+              partialWithdraw: () => {
+                return this
+              },
+            }
+            this.encodeABI = jest.fn(() => data)
+          }
+        }
+
+        walletService.web3.eth.Contract = ContractClass
+
+        walletService.partialWithdrawFromLock(lock, account, '3')
+
+        expect(walletService._sendTransaction).toHaveBeenCalledWith(
+          {
+            to: lock,
+            from: account,
+            data,
+            gas: 1000000,
```

Comment:
do we want this to fail if the gas cost changes? If so, keep it, otherwise I would use the value from `walletService.gasAmountConstants()`

Reply:
Good question! So we've actually added a line item for partial withdrawal costs in the constants (which is certainly wrong and may need to be adjusted, @hardlydifficult feel free to let me know if that needs a change). As for the hardcoded number, I did it based on the test for the full withdrawal which used a hardcoded number. I agree that the intent is not entirely clear there. I do plan on doing a refactoring pass on this file in a separate PR, maybe that can be included?

Reply:
sure. Your call on process, but let's get that sorted eventually. This PR is good to go if it isn't a quick fix

Reply:
`partialWithdraw` should require slightly more gas than `withdraw`, however they are very similar and should be much cheaper than the 1,000,000 limit you have ATM.

On a related note, #492 (use estimate gas) may change the solution here.

Reply:
Thanks for the feedback! I will do a follow-on test refactor PR to keep this one focused, and I'll keep my eye on #492